viewing_party/party.py

```diff
@@ -1,23 +1,179 @@
+from collections import Counter
+from shutil import move
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {
+        "title":title,
+        "genre":genre,
+        "rating":rating
+    }
+
+    for val in movie_dict.values():
+        if not val:
+            return None
+
+    return movie_dict
+
+def add_to_watched(user, movie):
+
+    for key in user:
+        user[key].append(movie)
+    
+    return user
+
+def add_to_watchlist(user, movie):
+
+    for key in user:
+        user[key].append(movie)
+    
+    return user
+
+def watch_movie(user_data, movie):
+    
+    filtered_user_data = dict(user_data)
+
+    for elem in user_data["watchlist"]:
+        watched_movie = elem["title"]
+        if movie == watched_movie:
+            filtered_user_data["watched"].append(elem)
+            filtered_user_data["watchlist"].remove(elem)
+
+    return filtered_user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    tot_rating = 0.0
+    
+    if not user_data["watched"]:
+        return tot_rating
+
+    for elem in user_data["watched"]:
+        tot_rating += elem["rating"]
+    
+    return tot_rating / len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+
+    if not user_data["watched"]:
+        return None
+
+    for elem in user_data["watched"]:
+        genre_list.append(elem["genre"])
+        
+    counters = Counter(genre_list) #returns dictionary with genre as key and number of times it appears as val

```

Comment:
Thanks for explicitly calling out what Counter() is returning. In the future, I'd like to see you use docstrings in your functions when you introduce a package like this. 

counters could be renamed to something like genre_frequencies for clarity.

Another way that this function could've been implemented is to loop through user_data["watched"] and populate a genre_frequency dictionary without using Counter and passing genre_frequency to max like you did on line 70.